File: Bailaconmigo/src/main/java/org/example/bailaconmigo/DTOs/DanceDiversityReportDto.java
package org.example.bailaconmigo.DTOs;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DanceDiversityReportDto {
    private Long totalActiveDancers;
    private Double averageStylesPerDancer;
    private List<DanceStyleDistributionDto> styleDistribution;
    private List<LevelDistributionDto> levelDistribution;
    private List<StyleCombinationDto> topCombinations;
    private List<RegionalDanceStyleDto> stylesByRegion;
    private List<StyleLevelStatsDto> styleLevelStats;
}

File: frontend/src/app/models/dance-diversity-report-dto.ts
import { DanceStyleDistributionDto } from "./dance-style-distribution-dto";
import { LevelDistributionDto } from "./level-distribution-dto";
import { RegionalDanceStyleDto } from "./regional-dance-style-dto";
import { StyleCombinationDto } from "./style-combination-dto";
import { StyleLevelStatsDto } from "./style-level-stats-dto";

export interface DanceDiversityReportDto {
  totalActiveDancers: number;
  averageStylesPerDancer: number;
  styleDistribution: DanceStyleDistributionDto[];
  levelDistribution: LevelDistributionDto[];
  topCombinations: StyleCombinationDto[];
  stylesByRegion: RegionalDanceStyleDto[];
  styleLevelStats: StyleLevelStatsDto[];
}
